test/process_test.cc

```diff
@@ -88,10 +126,17 @@ INSTANTIATE_TEST_SUITE_P(IpVersions, ProcessTest,
                          Envoy::TestUtility::ipTestParamsToString);
 
 TEST_P(ProcessTest, TwoProcessInSequence) {
+  FakeStatsSinkFactory factory;
```

Comment:
Instead of modifying the existing test case by adding another tested behavior, we would be better off defining a new test case for this new behavior that includes the stats sink. The general advice is to only test one behavior per test case, since test failures are then easier to understand.

Applies here and below.

Reply:
Done. Added new test cases for the new behavior